sdk/data/aztables/doc.go

```diff
@@ -0,0 +1,207 @@
+ensure proper round-trip serialization of all properties.
+
+	aztables.EDMEntity{
+		Entity: aztables.Entity{
+			PartitionKey: "pencils",
+			RowKey: "id-003",
+		},
+		Properties: map[string]interface{}{
+			"Product": "Ticonderoga Pencils",
+			"Price": 5.00,
+			"Count": aztables.EDMInt64(12345678901234),
+			"ProductGUID": aztables.EDMGUID("some-guid-value"),
+			"DateReceived": aztables.EDMDateTime(time.Date{....}),
+			"ProductCode": aztables.EDMBinary([]byte{"somebinaryvalue"})
+		}
+	}
+
+More Examples
+
+The following sections provide several code snippets covering some of the most common Table tasks, including:
+
+* Creating a table
+* Creating entities
+* Querying entities
+
+
+Creating a Table
+
+Create a table in your account and get a `Client` to perform operations on the newly created table:
+
+	cred, err := aztables.NewSharedKeyCredential("myAccountName", "myAccountKey")
+	handle(err)
+	service, err := aztables.NewServiceClient("https://<my_account_name>.table.core.windows.net", cred, nil)
+	handle(err)
+	resp, err := service.CreateTable("myTable")
+
+Creating Entities
+
+	cred, err := aztables.NewSharedKeyCredential("myAccountName", "myAccountKey")
+	handle(err)
+	service, err := aztables.NewServiceClient("https://<my_account_name>.table.core.windows.net", cred, nil)
+	handle(err)
+
+	myEntity := aztables.EDMEntity{
+		Entity: aztables.Entity{
+			PartitionKey: "001234",
+			RowKey: "RedMarker",
+		},
+		Properties: map[string]interface{}{
+			"Stock": 15,
+			"Price": 9.99,
+			"Comments": "great product",
+			"OnSale": true,
+			"ReducedPrice": 7.99,
+			"PurchaseDate": aztables.EDMDateTime(time.Date(2021, time.August, 21, 1, 1, 0, 0, time.UTC)),
+			"BinaryRepresentation": aztables.EDMBinary([]byte{"Bytesliceinfo"})
+		}
+	}
+	marshalled, err := json.Marshal(myEntity)
+	handle(err)
+
+	client, err := service.NewClient("myTable")
+	handle(err)
+
+	resp, err := client.AddEntity(context.Background(), marshalled, nil)
+	handle(err)
+
+Querying entities
+
+	cred, err := aztables.NewSharedKeyCredential("myAccountName", "myAccountKey")
+	handle(err)
+	client, err := aztables.NewClient("https://myAccountName.table.core.windows.net/myTableName", cred, nil)
+	handle(err)
+
+	filter := "PartitionKey eq 'markers' or RowKey eq 'id-001'"
+	options := &ListEntitiesOptions{
+		Filter: &filter,
+		Select: to.StringPtr("RowKey,Value,Product,Available"),
+		Top: to.Int32Ptr(15),
+	}
+
+	pager := client.List(options)
+	for pager.NextPage(context.Background()) {
+		resp := pager.PageResponse()
+		fmt.Printf("Received: %v entities\n", len(resp.Entities))
+
+		for _, entity := range resp.Entities {
+			var myEntity aztables.EDMEntity
+			err = json.Unmarshal(entity, &myEntity)
+			handle(err)
+
+			fmt.Printf("Received: %v, %v, %v, %v\n", myEntity.Properties["RowKey"], myEntity.Properties["Value"], myEntity.Properties["Product"], myEntity.Properties["Available"])
```

Comment:
Probably equivalent, but are there first-class properties for RowKey and PartitionKey in EDMEntity? Or do you need to get them from the map?